sdk/storage/src/datalake/path_client.cpp

```diff
@@ -437,28 +443,36 @@ namespace Azure { namespace Storage { namespace DataLake {
     protocolLayerOptions.Timeout = options.Timeout;
     auto result = DataLakeRestClient::Path::Read(
         m_dfsUri.ToString(), *m_pipeline, options.Context, protocolLayerOptions);
-    auto range = GetOffsetLength(result.ContentRange);
-    return ReadPathResponse{std::move(result.BodyStream),
-                            std::move(result.AcceptRanges),
-                            std::move(result.CacheControl),
-                            std::move(result.ContentDisposition),
-                            std::move(result.ContentEncoding),
-                            std::move(result.ContentLanguage),
-                            std::move(result.ContentLength),
-                            range.first,
-                            range.second,
-                            std::move(result.ContentType),
-                            std::move(result.ContentMD5),
-                            std::move(result.Date),
-                            std::move(result.ETag),
-                            std::move(result.LastModified),
-                            std::move(result.RequestId),
-                            std::move(result.Version),
-                            std::move(result.ResourceType),
-                            std::move(result.LeaseDuration),
-                            std::move(result.LeaseState),
-                            std::move(result.LeaseStatus),
-                            std::move(result.XMsContentMd5),
-                            Details::DeserializeMetadata(result.Properties)};
+    Azure::Core::Nullable<int64_t> RangeOffset;
+    Azure::Core::Nullable<int64_t> RangeLength;
+    if (result.ContentRange.HasValue())
+    {
+      auto range = GetOffsetLength(result.ContentRange.GetValue());
```

Comment:
When will `range.second` be null?

Reply:
When `result.ContentRange` is null for the response and when customer specifies `bytes=<somevalue>-` for requests.

Reply:
I doubt that

> Host: adlsgen2account.dfs.core.windows.net
> User-Agent: curl/7.54.0
> Accept: */*
> Range: bytes=1-
> 
< HTTP/1.1 206 Partial Content
< Content-Length: 11
< Content-Type: application/octet-stream
< Content-Range: bytes 1-11/12
< Last-Modified: Thu, 02 Jul 2020 15:37:28 GMT
< Accept-Ranges: bytes
< ETag: "0x8D81E9DD3983864"
< Server: Windows-Azure-HDFS/1.0 Microsoft-HTTPAPI/2.0
< x-ms-properties: 
< x-ms-resource-type: file
< x-ms-lease-state: available
< x-ms-lease-status: unlocked
< x-ms-server-encrypted: false
< x-ms-content-md5: b1kCrCNwJL3QwXbLkwY9xA==
< x-ms-request-id: 60e167a9-d01f-0022-0e86-50aade000000
< x-ms-version: 2019-10-10
< Date: Thu, 02 Jul 2020 15:38:26 GMT
<